lib/services/ios-debug-service.ts

```diff
@@ -202,24 +201,34 @@ class IOSDebugService implements IDebugService {
                         } catch (e) {
                             this.$errors.failWithoutHelp(`The application ${projectId} timed out when performing the NativeScript debugger handshake.`);
                         }
-                        this.readyForAttachAction(iosDevice).wait();
+                        this.wireDebuggerClinet(() => iosDevice.connectToPort(InspectorBackendPort)).wait();
                         break;
                     case readyForAttach:
-                        this.readyForAttachAction(iosDevice).wait();
+                        this.wireDebuggerClinet(() => iosDevice.connectToPort(InspectorBackendPort)).wait();
                         break;
                 }
             }).future<void>()()).wait();
         }).future<void>()();
     }
+    
+    private wireDebuggerClinet(factory: () => net.Socket): IFuture<void> {
+        return (() => {
+            let frameworkVersion = this.getProjectFrameworkVersion().wait();
+            let socketFileLocation = "";
+            if (semver.gte(frameworkVersion, "1.4.0")) {
+                socketFileLocation = temp.path({ suffix: ".sock" });
+                createTcpSocketProxy(socketFileLocation, this.$logger, (callback) => connectEventually(factory, callback));
```

Comment:
Isn't it better if we return the socket file descriptor from this method instead of passing it as first parameter?